source/includes/fundamentals/code-snippets/MonolightCodecProvider.java

```diff
@@ -0,0 +1,35 @@
+package fundamentals.monolightcodec;
+
+import org.bson.codecs.Codec;
+import org.bson.codecs.IntegerCodec;
+import org.bson.codecs.configuration.CodecProvider;
+import org.bson.codecs.configuration.CodecRegistry;
+
+
+// start class
+public class MonolightCodecProvider implements CodecProvider {
+    private MonolightCodec monolightCodec;
+    private IntegerCodec integerCodec;
+    private PowerStatusCodec powerStatusCodec;
+
+    public MonolightCodecProvider() {
+        integerCodec = new IntegerCodec();
+        powerStatusCodec = new PowerStatusCodec();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> Codec<T> get(Class<T> clazz, CodecRegistry registry) {
+        if (clazz == Monolight.class) {
+            return (Codec<T>) new MonolightCodec(registry);
+        } else if (clazz == Integer.class) {
+            return (Codec<T>) integerCodec;
+        } else if (clazz == String.class) {
+            return (Codec<T>) powerStatusCodec;
```

Comment:
This should be treated with caution - as it will handle *all* string values.  

So if a user was using the codec registry to decode a `Document` then any string values would encoded using the powerStatusCodec!

Reply:
I figured that would happen, but is there a convention that I can use to selectively encode strings? Would I need to create a conditional that checked the value of the string to determine whether to use a PowerStatusCodec or a StringCodec?

Reply:
The CodecRegistry is a global registry. So the convention would probably be for the MonolightCodec would handle the power status conversion internally.